src/LatencyEstimator.cpp

```diff
@@ -37,6 +38,11 @@ LatencyEstimator::LatencyEstimator(const Config& config)
     , output_trigger_(config_)
     , input_trigger_(config_)
     , sma_(config.sma_window) {
+        config_.enable_json ? format = new JSONFormatter : format = new TextFormatter;
```

Comment:
Instead of duplicating this check in LatencyEstimator and LossEstimator, I suggest to do it once in main() and then just pass IFormatter to estimator constructor. Since formatter should be always present, it may be a reference, not a pointer.